src/library-authoring/LibraryBlock/LibraryBlock.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useRef, useState } from 'react';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { getConfig } from '@edx/frontend-platform';
+
+import messages from './messages';
+
+interface LibraryBlockProps {
+  onBlockNotification?: (event: { eventType: string; [key: string]: any }) => void;
+  usageKey: string;
+}
+/**
+ * React component that displays an XBlock in a sandboxed IFrame.
+ *
+ * The IFrame is resized responsively so that it fits the content height.
+ *
+ * We use an IFrame so that the XBlock code, including user-authored HTML,
+ * cannot access things like the user's cookies, nor can it make GET/POST
+ * requests as the user. However, it is allowed to call any XBlock handlers.
+ */
+const LibraryBlock = ({ onBlockNotification, usageKey }: LibraryBlockProps) => {
+  const iframeRef = useRef<HTMLIFrameElement>(null);
+  const [iFrameHeight, setIFrameHeight] = useState(600);
+  const lmsBaseUrl = getConfig().LMS_BASE_URL;
+
+  const intl = useIntl();
+
+  /**
+   * Handle any messages we receive from the XBlock Runtime code in the IFrame.
+   * See wrap.ts to see the code that sends these messages.
+   */
+  /* istanbul ignore next */
+  const receivedWindowMessage = async (event) => {
+    if (!iframeRef.current || event.source !== iframeRef.current.contentWindow) {
+      return; // This is some other random message.
+    }
+
+    const { method, replyKey, ...args } = event.data;
+
+    if (method === 'update_frame_height') {
+      setIFrameHeight(args.height);
+    } else if (method?.indexOf('xblock:') === 0) {
+      // This is a notification from the XBlock's frontend via 'runtime.notify(event, args)'
+      if (onBlockNotification) {
+        onBlockNotification({
+          eventType: method.substr(7), // Remove the 'xblock:' prefix that we added in wrap.ts
+          ...args,
+        });
+      }
+    }
+  };
+
+  /**
+   * Load the XBlock data from the LMS and then inject it into our IFrame.
```

Comment:
```suggestion
   * Prepare to receive messages from the IFrame.
```
We're not "injecting" the XBlock HTML/data anymore - the LMS is now rendering it.